lib/src/select/Select.test.js

```diff
@@ -293,26 +278,49 @@ describe("Select component tests", () => {
     expect(getByText("Option 02, Option 03, Option 04, Option 06")).toBeTruthy();
     expect(submitInput.value).toBe("4,2,6,3");
   });
-  test("Disabled select - Clear all options action must be shown but not clickable", () => {
-    const { getByRole, getByText, queryByRole } = render(
-      <DxcSelect label="test-select-label" value={["1", "2"]} options={single_options} disabled searchable multiple />
+  test("Disabled select - Listbox cannot be opened by click or keyboard press", () => {
+    const { getByRole, queryByRole } = render(
+      <DxcSelect label="test-select-label" value={["1", "2"]} options={single_options} disabled />
     );
     const select = getByRole("combobox");
 
     expect(select.getAttribute("aria-disabled")).toBe("true");
     userEvent.click(select);
     expect(queryByRole("listbox")).toBeFalsy();
+    fireEvent.keyDown(select, { key: "ArrowUp", code: "ArrowUp", keyCode: 38, charCode: 38 });
+    expect(queryByRole("listbox")).toBeFalsy();
+    fireEvent.keyDown(select, { key: "ArrowDown", code: "ArrowDown", keyCode: 40, charCode: 40 });
+    expect(queryByRole("listbox")).toBeFalsy();
+  });
+  test("Disabled select - Clear all options action must be shown but not clickable", () => {
+    const { getByRole, getByText } = render(
+      <DxcSelect label="test-select-label" value={["1", "2"]} options={single_options} disabled searchable multiple />
+    );
+
     userEvent.click(getByRole("button"));
     expect(getByText("Option 01, Option 02")).toBeTruthy();
   });
-  test("Focused select does not open the listbox", () => {
+  test("Disabled select - Does not call onBlur event", () => {
+    const onBlur = jest.fn();
+    const { getByRole } = render(
+      <DxcSelect label="test-select-label" options={single_options} disabled onBlur={onBlur} />
+    );
+    const select = getByRole("combobox");
+
+    userEvent.click(select);
+    fireEvent.keyDown(getByRole("combobox"), { key: "Tab", code: "Tab", keyCode: 9, charCode: 9 });
+    expect(onBlur).not.toHaveBeenCalled();
+  });
+  test("Disabled select - When the component gains the focus, the listbox does not open", () => {
```

Comment:
The select does not open when focused even when it's not disabled. We should test that the select can not gain the focus when disabled.
Also, the click test is repeated by the test in line 281.

Reply:
You are right Jiale! Already change it 👍🏼.